console/src/main/java/org/nervos/ckb/example/TransactionExample.java

```diff
@@ -0,0 +1,145 @@
+      throw new Exception("No enough Capacities");
+    }
+    List<CellOutput> cellOutputs = new ArrayList<>();
+    receiverBeanList.forEach(
+        (receiverBean -> {
+          AddressUtils addressUtils = new AddressUtils(Network.TESTNET);
+          String blake2b = addressUtils.getBlake160FromAddress(receiverBean.address);
+          cellOutputs.add(
+              new CellOutput(
+                  receiverBean.capacity,
+                  "0x",
+                  new Script(systemContract.systemScriptCellHash, Arrays.asList(blake2b))));
+        }));
+    if (cellInputsAndBalanceSum.capacity.compareTo(needCapacities) > 0) {
+      cellOutputs.add(
+          new CellOutput(
+              cellInputsAndBalanceSum.capacity.subtract(needCapacities).toString(10),
+              "0x",
+              inputLockScript));
+    }
+    Transaction transaction =
+        new Transaction(
+            "0",
+            Arrays.asList(new OutPoint(null, systemContract.systemScriptOutPoint)),
+            cellInputsAndBalanceSum.inputs,
+            cellOutputs,
+            new ArrayList<>());
+    String txHash = ckbService.computeTransactionHash(transaction).send().getTransactionHash();
+    Witness witness = new Witness(Numeric.toBigInt(privateKeyHex), txHash);
+    cellInputsAndBalanceSum.inputs.forEach(
+        cellInput -> {
+          transaction.witnesses.add(witness);
+        });
+    return transaction;
+  }
+
+  private CellInputsAndBalanceSum generateInputs(String lockHash, BigInteger needCapacities)
+      throws Exception {
+    List<CellInput> cellInputs = new ArrayList<>();
+    BigInteger inputsCapacities = BigInteger.ZERO;
+    long toBlockNumber = ckbService.getTipBlockNumber().send().getBlockNumber().longValue();
+    long fromBlockNumber = 1;
+    while (fromBlockNumber <= toBlockNumber && inputsCapacities.compareTo(needCapacities) < 0) {
+      long currentToBlockNumber = Math.min(fromBlockNumber + 100, toBlockNumber);
+      List<CellOutputWithOutPoint> cellOutputs =
+          ckbService
+              .getCellsByLockHash(
+                  lockHash, String.valueOf(fromBlockNumber), String.valueOf(currentToBlockNumber))
+              .send()
+              .getCells();
+
+      if (cellOutputs != null && cellOutputs.size() > 0) {
+        for (CellOutputWithOutPoint cellOutputWithOutPoint : cellOutputs) {
+          CellInput cellInput =
+              new CellInput(cellOutputWithOutPoint.outPoint, Collections.emptyList(), "0");
+          inputsCapacities =
+              inputsCapacities.add(new BigDecimal(cellOutputWithOutPoint.capacity).toBigInteger());
+          cellInputs.add(cellInput);
+          if (inputsCapacities.compareTo(needCapacities) > 0) {
+            break;
+          }
+        }
+      }
+      fromBlockNumber = currentToBlockNumber + 1;
+    }
+    return new CellInputsAndBalanceSum(cellInputs, new BigDecimal(inputsCapacities).toBigInteger());
+  }
+
+  class CellInputsAndBalanceSum {
```

Comment:
CellInputs is OK. Why name is so long?

Reply:
There is no balance in CellInput,so 'CellInputs' can not describe this class. CellInputs + Balance

Reply:
No need to describe all inforamation